core/org.wso2.carbon.base/src/test/java/org/wso2/carbon/base/ServerConfigurationTest.java

```diff
@@ -0,0 +1,185 @@
+                expectedProperties, actualProperties1);
+        String actualProperties2[] = ServerConfiguration.getInstance().getProperties("NotPresentInConfig");
+        assertEquals(actualProperties2.length, 0, "Testing properties received for a key not present in config");
+    }
+
+    @Test(groups = {"org.wso2.carbon.base"})
+    public void testGetDocumentElement() throws XMLStreamException, ClassNotFoundException, IllegalAccessException,
+            InstantiationException, NoSuchMethodException, InvocationTargetException, ParserConfigurationException,
+            IOException, SAXException {
+        InputStream carbonInputStream = readFile("carbon.xml");
+        OMElement documentElement = new StAXOMBuilder(carbonInputStream)
+                .getDocumentElement();
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        documentElement.serialize(outputStream);
+        ByteArrayInputStream inputStream = new ByteArrayInputStream(
+                outputStream.toByteArray());
+
+        DocumentBuilderFactory factory = getSecuredDocumentBuilder();
+        factory.setNamespaceAware(true);
+        Element expectedElement = factory.newDocumentBuilder().parse(inputStream)
+                .getDocumentElement();
+        Element actualElement = ServerConfiguration.getInstance().getDocumentElement();
+        assertEquals(actualElement.toString(), expectedElement.toString());
+    }
+
+    /**
+     * Used for testGetDocumentElement.
+     *
+     * @return a DocumentBuilderFactory instance
+     */
+    private DocumentBuilderFactory getSecuredDocumentBuilder() throws ParserConfigurationException {
+
+        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+        dbf.setNamespaceAware(true);
+        dbf.setXIncludeAware(false);
+        dbf.setExpandEntityReferences(false);
+
+        dbf.setFeature(Constants.SAX_FEATURE_PREFIX + Constants.EXTERNAL_GENERAL_ENTITIES_FEATURE, false);
+        dbf.setFeature(Constants.SAX_FEATURE_PREFIX + Constants.EXTERNAL_PARAMETER_ENTITIES_FEATURE, false);
+        dbf.setFeature(Constants.XERCES_FEATURE_PREFIX + Constants.LOAD_EXTERNAL_DTD_FEATURE, false);
+
+        SecurityManager securityManager = new SecurityManager();
+        securityManager.setEntityExpansionLimit(0);
+        dbf.setAttribute(Constants.XERCES_PROPERTY_PREFIX + Constants.SECURITY_MANAGER_PROPERTY, securityManager);
+
+        return dbf;
+    }
+
+    /**
+     * This is used to get the contents of the txt file as an input stream
+     *
+     * @param path location path to the relevant txt file
+     * @return an inputstream containing the contents of the given file
+     */
+    private static InputStream readFile(String path) {
+        InputStream inputStream;
+        ClassLoader classLoader = ServerConfigurationTest.class.getClassLoader();
+        inputStream = classLoader.getResourceAsStream(path);
+        return inputStream;
```

Comment:
We can write this code as,

```java
ClassLoader classLoader = ServerConfigurationTest.class.getClassLoader();
return classLoader.getResourceAsStream(path);
```
IMO this would be more readable